### PR TITLE
RHIDP-1908: Documenting the telemetry feature in admin guide

### PR DESCRIPTION
**Pull request by shipsing**
**Purpose**: To resolve this issue: https://issues.redhat.com/browse/RHIDP-1908
SME review: @kadel @kim-tsao 
QE review: @zdrapela 
Peer review: @abrennan89 
PR preview: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-212/admin-rhdh/#assembly-rhdh-telemetry_admin-rhdh

Related: https://issues.redhat.com/browse/RHIDP-58

### CODE REVIEW

**Review by linfraze — COMMENTED** [review]
Replaced company name with the new attribute value

File: admin/assembly-rhdh-telemetry.adoc
Comment (kim-tsao) [inline admin/assembly-rhdh-telemetry.adoc:13]: Maybe we can refer to this as "system related information" rather than user?

File: admin/proc-enabling-telemetry.adoc
Comment (kim-tsao) [inline admin/proc-enabling-telemetry.adoc:14]: Same comment as above re: use package name instead

File: admin/proc-customizing-telemetry-destination.adoc
Comment (kim-tsao) [inline admin/proc-customizing-telemetry-destination.adoc:5]: 1.  Let's use "Segment source" instead of "new destination".  This is the terminology Segment uses (see https://segment.com/docs/connections/find-writekey/)

2.  Fix capitalization on segment and clarify it's a write key 

`However, you can configure a new Segment source that receives telemetry data based on your needs. For configuration, you need a unique Segment write key that points to the Segment source.`


Since we are allowing customers to instrument their own Segment service to collect data, this will be the same set of data that's specified in `4. Telemetry data collection`.  Customers may need to write up their own data collection notice. Can we somehow convey to them that they can expect to see the same set data?

File: admin/proc-customizing-telemetry-destination.adoc
Comment (kim-tsao) [inline admin/proc-customizing-telemetry-destination.adoc:23]: fix typo, it should be `upstream`

File: admin/proc-customizing-telemetry-destination.adoc
Comment (kim-tsao) [inline admin/proc-customizing-telemetry-destination.adoc:31]: Edit:

`<segment_key>` denotes a unique identifier for your Segment source.

File: admin/proc-customizing-telemetry-destination.adoc
Comment (kim-tsao) [inline admin/proc-customizing-telemetry-destination.adoc:47]: Edit:

<segment_key> denotes a unique identifier for your Segment source.

File: admin/assembly-rhdh-telemetry.adoc
Comment (shipsing) [reply admin/assembly-rhdh-telemetry.adoc:6]: In the dynamic plugins support matrix, the plugin is referred to as `@janus-idp/backstage-plugin-analytics-provider-segment`. I see your concern here. let's use the short name and provide some context about the plugin.

File: admin/proc-customizing-telemetry-destination.adoc
Comment (shipsing) [reply admin/proc-customizing-telemetry-destination.adoc:5]: @kim-tsao,  Can we mention the same in a note in this manner:
NOTE:
By configuring a new Segment source, you can collect and analyze the same set of data that is mentioned in the link:{LinkAdminGuide}#assembly-rhdh-telemetry_admin-rhdh[Telemetry data collection] section. You might also require to create your own telemetry data collection notice for your application users.

**Review by zdrapela — APPROVED** [review]
LGTM 👍️ 
Thanks!

File: admin/assembly-rhdh-telemetry.adoc
Comment (kim-tsao) [inline admin/assembly-rhdh-telemetry.adoc:21]: I think it's just the usernames that are used to generate the unique identifier.  We can remove `anonymized IP addresses`

File: admin/assembly-rhdh-telemetry.adoc
Comment (kim-tsao) [inline admin/assembly-rhdh-telemetry.adoc:29]: Do you mean users can enable telemetry in an air-gapped environment or would they need to disable it?

File: admin/proc-customizing-telemetry-segment.adoc
Comment (kim-tsao) [inline admin/proc-customizing-telemetry-segment.adoc:18]: `Configure integration to your Segment source...`

File: admin/assembly-rhdh-telemetry.adoc
Comment (shipsing) [reply admin/assembly-rhdh-telemetry.adoc:29]: @kim-tsao, Users can perform any of the actions (enable, disable)based on their needs. The intent is to let the users know that the configuration procedures are same even if they have installed RHDH in an airgapped envornment. the procedures do not vary in any manner.

Tweaked the note in this manner:
You can also customize the telemetry data collection feature and the telemetry Segment source configuration in an air-gapped environment. For more information, see the following procedures.

File: admin/assembly-rhdh-telemetry.adoc
Comment (kadel) [inline admin/assembly-rhdh-telemetry.adoc:19]:
```suggestion
* The usernames are recorded as an anonymized hash.   
```

File: admin/assembly-rhdh-telemetry.adoc
Comment (kadel) [inline admin/assembly-rhdh-telemetry.adoc:21]:
```suggestion
The username hash is a unique identifier that is used solely to identify the number of unique users of the {product-very-short} application.
```
hash is not random

File: installation/proc-install-rhdh-airgapped-environment-ocp-helm.adoc
Comment (kadel) [inline installation/proc-install-rhdh-airgapped-environment-ocp-helm.adoc:12]:
```suggestion 
```
telemetry is usually not used in air-gapped environments as the nature of air-gap environment makes in impossible to send data out which would defeat the pursue of air-gap

File: admin/proc-disabling-telemetry-in-local-environment.adoc
Comment (kadel) [inline admin/proc-disabling-telemetry-in-local-environment.adoc:1]: we can remove this whole section, RHDH is almost never used locally

File: admin/proc-disabling-telemetry.adoc
Comment (kadel) [inline admin/proc-disabling-telemetry.adoc:4]:
```suggestion
To disable telemetry data collection, you must disable the `analytics-provider-segment` plugin either using the Helm Chart or the {product} Operator configuration. 
```

File: admin/proc-disabling-telemetry-in-CI-environment.adoc
Comment (kadel) [inline admin/proc-disabling-telemetry-in-CI-environment.adoc:1]: We can remove this section from RHDH docs. THis makes sense only for upstream (showcase)

File: admin/proc-enabling-telemetry.adoc
Comment (kadel) [inline admin/proc-enabling-telemetry.adoc:11]: Lets remove this whole section. Those prereqs doesn't make much sense. This is part of the admin documentation, from that it should be clear that this is for RHDH administrators.

File: admin/proc-disabling-telemetry.adoc
Comment (kadel) [inline admin/proc-disabling-telemetry.adoc:11]: Lets remove this whole section. Those prereqs doesn't make much sense. This is part of the admin documentation, from that it should be clear that this is for RHDH administrators.

File: admin/proc-disabling-telemetry.adoc
Comment (kadel) [inline admin/proc-disabling-telemetry.adoc:17]:
```suggestion
. Disable the `analytics-provider-segment` plugin by using one of the following options:
```

File: admin/assembly-rhdh-telemetry.adoc
Comment (kadel) [inline admin/assembly-rhdh-telemetry.adoc:37]: Telemetery will be enabled by default, I think that it makes more sense to have this listed before enabling telemetry section

**Review by kadel — CHANGES_REQUESTED** [review]

File: admin/assembly-rhdh-telemetry.adoc
Comment (shipsing) [reply admin/assembly-rhdh-telemetry.adoc:8]: I understand that this is a sensitive feature and we cannot make false promises. I completely agree with you.

File: admin/assembly-rhdh-telemetry.adoc
Comment (shipsing) [reply admin/assembly-rhdh-telemetry.adoc]: @kadel @kim-tsao, Should we close [this ticket](https://issues.redhat.com/browse/RHIDP-1909), wherein we had to capture the procedure to disable telemetry in an airgap scenario? Because it is impractical to use telemetry in air-gapped environments.

**Review by abrennan89 — COMMENTED** [review]
Left a few comments; some are required changes, some are suggestions. Overall nice work!

One thing to note across the docs - per the IBM style guide, we use "plug-in" rather than "plugin". See https://www.ibm.com/docs/en/ibm-style?topic=word-usage#plug-in

File: assemblies/assembly-rhdh-telemetry.adoc
Comment (shipsing) [reply assemblies/assembly-rhdh-telemetry.adoc:27]: yes, it is the name of a component in context of Telemetry and the security team emphasized to keep the alphabet S in capital.

File: modules/admin/proc-customizing-telemetry-segment.adoc
Comment (shipsing) [reply modules/admin/proc-customizing-telemetry-segment.adoc:9]: yes, the link will not work now because the telemetry feature is going to be published with the RHDH 1.2 release.

**Review by kadel — APPROVED** [review]

**Review by kadel — APPROVED** [review]
This looks good. Great work @shipsing 🎉

File: assemblies/assembly-rhdh-telemetry.adoc
Comment (nickboldt) [inline assemblies/assembly-rhdh-telemetry.adoc:4]: I don't think we should have this on by default until we have a documented approval from Legal. https://issues.redhat.com/browse/RHIDP-58?focusedId=24806334&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24806334 contains no such documentation.

**Review by nickboldt — CHANGES_REQUESTED** [review]
Setting this to blocked because we don't have DOCUMENTED approval from Legal to enable an opt-out telemetry feature. 

If we change the code (and therefore the doc) to opt-in (disabled by default) then I'm otherwise happy with this PR.

**Review by zdrapela — APPROVED** [review]

File: assemblies/assembly-rhdh-telemetry.adoc
Comment (nickboldt) [inline assemblies/assembly-rhdh-telemetry.adoc:4]:
```suggestion
The telemetry data collection feature is disabled by default. If enabled, {product} will send telemetry data to Red Hat by using the `backstage-plugin-analytics-provider-segment` plugin.
```

Comment (abrennan89) [reply assemblies/assembly-rhdh-telemetry.adoc:4]: Per the RH supplementary style guide (and IIRC, maybe the IBM one too) we actually discourage using the future tense unless absolutely necessary, so I'd leave this as is, or "If the feature is enabled, {product} sends telemetry data...". See https://redhat-documentation.github.io/supplementary-style-guide/#will

Comment (nickboldt) [reply assemblies/assembly-rhdh-telemetry.adoc:4]: :as-you-wish:

**Review by nickboldt — APPROVED** [review]

File: assemblies/assembly-rhdh-telemetry.adoc
Comment (nickboldt) [inline assemblies/assembly-rhdh-telemetry.adoc:4]:
```suggestion
The telemetry data collection feature is enabled by default. {product} sends telemetry data to Red Hat by using the `backstage-plugin-analytics-provider-segment` plugin.
```

**Comment by kadel** [comment]
We can go ahead and merge this. Just to confirm, we are going with enabled by default (opt-out) more information in https://issues.redhat.com/browse/RHIDP-1311 and the related parent feature

**Comment by nickboldt** [comment]
Merging as we have approvals from at least a few people.

**Pull request merged**
